routemaster/jobs/batch.rb

```diff
@@ -35,21 +36,15 @@ def call(uid)
           batch.delete
         rescue Services::Deliver::CantDeliver => e
           _log_exception(e)
-          attempts = batch.fail
-          raise Models::Queue::Retry, _backoff(attempts)
+          raise Models::Queue::Retry, _backoff(batch)
         end
         self
       end
 
       private
       
-      def _backoff(attempts)
-        backoff = 1_000 * 2 ** [attempts-1, _backoff_limit].min
-        backoff + rand(backoff)
-      end
-
-      def _backoff_limit
-        @@_backoff_limit ||= Integer(ENV.fetch('ROUTEMASTER_BACKOFF_LIMIT'))
+      def _backoff(batch)
+        Services::Throttle.new(batch: batch).retry_backoff
```

Comment:
You end up needing to re-instantiate `Throttle` after the exception was raised, which means you end up re-calculating a delay which was just done (which might cause more Redis calls, as you need to load the subscriber data again from the batch).

I'd suggest instead adding a `delay` field to the `CantDeliver` exception and simply using that (ie. keep all interactions with throttling/delay math under the `Deliver` service)